mesonbuild/environment.py

```diff
@@ -888,8 +888,8 @@ def _guess_nix_linker(self, compiler: T.List[str], comp_class: T.Type[Compiler],
             check_args += override
 
         _, o, e = Popen_safe(compiler + check_args)
-        v = search_version(o)
-        if o.startswith('LLD'):
+        v = search_version(o + e)
+        if o.startswith('LLD') or 'LLVM' in e:
```

Comment:
This seems off to me. If qualcom sees fit to not identify their linker as lld, i'd rather have a separate class for it, even if it inherits the lld class.

Reply:
Yes please. I don't know Meson enough to write up a separate class for baremetal LLVM linker variants though. Could you please pass me some pseudo-code to start with?

Reply:
> Does this linker actually work with gnu? I think we probably want to put this in the clang implementation.

Oh. That explains why Meson (w/o my patch) classified the Qualcomm linker as Apple. Again, I am happy to rewrite the patch if you can pass me some pseudo code of the derived class.

Reply:
Oh. That explains why Meson (w/o my patch) classified the Qualcomm linker as Apple's clang. Again, I am happy to rewrite the patch if you can pass me some pseudo code of the derived class.

Reply:
What I'd do here then is something like (in mesonbuild/linkers.py):

```python
class QualcommLLVMDynamicLinker(LLVMDynamicLinker):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.id = 'ld.qcld'  # or you could refactor LLVMDynamicLinker to not hardcode ld.lld
```

Then you need to add it to the all list and do the proper finding thing for it in env.